### ISSUE TITLE
Remove Plugin Config when we delete Teams and remove orphaned teams

### ISSUE THREAD

**Issue by fuziontech** [PostHog/posthog#13595]
Noticed these logs coming from the async plugin service:
https://grafana.us-east-1.prod.posthog.dev/explore?orgId=1&left=%7B%22datasource%22:%22P8E80F9AEF21F6940%22,%22queries%22:%5B%7B%22refId%22:%22A%22,%22editorMode%22:%22builder%22,%22expr%22:%22%7Bcontainer%3D%5C%22posthog-plugins-exports%5C%22%7D%20%7C%3D%20%60without%20team_id%21%60%22,%22queryType%22:%22range%22%7D%5D,%22range%22:%7B%22from%22:%22now-1h%22,%22to%22:%22now%22%7D%7D

Basically, currently we are creating orphaned `PluginConfigs` when a team is deleted which is not very tidy and results in a bit of log spam. We should do a few things to clean this up.

1. Create a migration to cascade delete `posthog_pluginconfig` on team delete
2. Delete already orphaned plugin configs that we have

**Comment by tomasfarias** [comment]
There are a few reasons that I'm looking into to explain why orphaned `PluginConfig`s are popping up:

1. The `PluginConfig` Django model already has a `on_delete=models.CASCADE` setting. However, this is not the same as a `ON CASCADE DELETE` at the database level: instead, Django emulates the behavior. In other words, if we run a delete query manually (e.g. `DELETE FROM Team WHERE ...`) Django won't delete the `PluginConfig`.
2. `PluginConfig` does allow `NULL` values as the `Team` foreign key. So we may just be writing orphaned `PluginConfig`s from the get go.

**Comment by tomasfarias** [comment]
Update: I've confirmed there is only one `PluginConfig` affected (id=17357):
* It's the only one that appears in the logs.
* A quick DB query revealed it's the only with `team_id=0`.

**Comment by tomasfarias** [comment]
Resolution: since `team_id=0` is a demo account, and we don't want plugins running for a demo account, the easiest way to resolve this is just deleting the plugin config. This has been executed :+1: 

That being said, we are relying on a falsy check to determine if a team has plugins to be loaded: https://github.com/PostHog/posthog/blob/master/plugin-server/src/worker/plugins/loadPluginsFromDB.ts#L54

This produces a false positive with `team_id=0` as 0 is falsy, even though the team exists. This check should check for `null` and a different mechanism should be used to check for demo accounts (`is_demo` flag for example).
